### ISSUE TITLE
prefer-method-signature with inline type

### ISSUE THREAD

**Issue by stephenlautier** [palantir/tslint#2414]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.2.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
export function registerGulpMultiTargetBuilds(options: {
	taskName: string,
	action: (target: string) => Promise<any>,
	config: { devTarget: string, buildTargets: string[] }
}) {
	...
};
```

with `tslint.json` configuration:

```json
{
	"rules": {
		"prefer-method-signature": true
	}
}
```

#### Actual behavior
"Inline typings" e.g. 

```ts
options: {
	action: (target: string) => Promise<any>,
}
```
are currently also being linted

```txt
prefer-method-signature  Use a method signature instead of a property signature of function type.
```

#### Expected behavior
When using "inline typings" shouldn't suggest using method signature as it will cause some other lint to fail such as unused variable (parameter). Also generally when functions are passed as params are passed more as properties rather than methods (i might be wrong)

**Comment by andy-hanson** [comment]
Using `action(target: string): Promise<any>;` won't cause unused-parameter errors. It's just a syntax choice, nothing more.

**Comment by stephenlautier** [comment]
@andy-hanson i think you're assuming not because you've tried it.

![2017-03-28 20_43_39- gulp-utils ts - ssv-tools - visual studio code](https://cloud.githubusercontent.com/assets/3908723/24421692/505a8b3a-13f7-11e7-9b37-fce280734112.png)

its also triggering `member-ordering` below it (but thats oki)

**Comment by andy-hanson** [comment]
That's a bug in that lint rule then. TypeScript's own [`--noUnusedParameters`](https://www.typescriptlang.org/docs/handbook/compiler-options.html) doesn't warn for that.
Should be fixed by #2235, which uses TypeScript's implementation now.

**Comment by JoshuaKGoldberg** [comment]
> Should be fixed by #2235

🚀
